Ligo/stablecoin/actions.ligo

```diff
@@ -0,0 +1,516 @@
+      | None ->
+          failwith ("Spender address is not present in owner's minter list")
+      end
+  | None -> failwith ("Owner address is not present in ledger")
+  end
+} with (upd, ups)
+
+/* ------------------------------------------------------------- */
+
+/*
+ * Entrypoints
+ */
+
+/*
+ * Transfers given amount of tokens between addresses
+ *
+ * parameter is list of record
+ * [ batch    : list (transferDescriptor)
+ * ; operator : address
+ * ]
+ */
+function transfer
+  ( const parameter : transferParams
+  ; const store     : storage
+  ) : entrypoint is block
+{
+  // Remnants of fa1.2 interface
+  //const debitParams : burnParams = record
+  //  [ fromAddress = parameter.fromAddress
+  //  ; amount      = parameter.amount
+  //  ];
+  //const creditParams : mintParams = record
+  //  [ toAddress = parameter.toAddress
+  //  ; amount    = parameter.amount
+  //  ];
+
+  // const ep1 : entrypoint = debitFrom (debitParams  , store);
+  // const ep2 : entrypoint = creditTo  (creditParams , store);
+
+  const upd : list (operation) = nil;
+  const ups : storage = store;
+} with (upd, ups)
+
+/*
+ * TODO
+ */
+function balanceOf
+  ( const parameter : balanceOfParams
+  ; const store     : storage
+  ) : entrypoint is block
+{
+  const upd : list (operation) = nil;
+  const ups : storage = store;
+} with (upd, ups)
+
+/*
+ * TODO
+ */
+function getTotalSupply
+  ( const parameter : getTotalSupplyParams
+  ; const store     : storage
+  ) : entrypoint is block
+{
+  const upd : list (operation) = nil;
+  const ups : storage = store;
+} with (upd, ups)
+
+/*
+ * TODO
+ */
+function getTokenMetadata
+  ( const parameter : getTokenMetadataParams
+  ; const store     : storage
+  ) : entrypoint is block
+{
+  const upd : list (operation) = nil;
+  const ups : storage = store;
+} with (upd, ups)
+
+/*
+ * TODO
+ */
+function getPermissionsDescriptor
+  ( const parameter : getPermissionsDescriptorParams
+  ; const store     : storage
+  ) : entrypoint is block
+{
+  const upd : list (operation) = nil;
+  const ups : storage = store;
+} with (upd, ups)
+
+/*
+ * TODO
+ */
+function setTransferHook
+  ( const parameter : setTransferHookParams
+  ; const store     : storage
+  ) : entrypoint is block
+{
+  const upd : list (operation) = nil;
+  const ups : storage = store;
+} with (upd, ups)
+
+/*
+ * TODO
+ */
+function updateOperators
+  ( const parameter : updateOperatorParams
+  ; const store     : storage
+  ) : entrypoint is block
+{
+  const upd : list (operation) = nil;
+  const ups : storage = store;
+} with (upd, ups)
+
+/*
+ * TODO
+ */
+function isOperator
+  ( const parameter : isOperatorParams
+  ; const store     : storage
+  ) : entrypoint is block
+{
+  const upd : list (operation) = nil;
+  const ups : storage = store;
+} with (upd, ups)
+
+/*
+ * Enables spender to withdraw given amount of tokens from a sender
+ *
+ * parameter is record
+ * [ spender : address
+ * ; amount  : nat
+ * ]
+ */
+function approve
```

Comment:
I don't see `approve` in FA2. Is it a leftover or is it actually needed?

Reply:
AFAIU from the specification document in this repo, the `approve` is replaced by `addOperators` and `approveOperatorUpdate` operations. Right @awkure ?